public/app/features/search/service/folders.ts

```diff
@@ -0,0 +1,52 @@
+import { getBackendSrv } from '@grafana/runtime';
+import config from 'app/core/config';
+
+import { DashboardViewItem } from '../types';
+
+import { getGrafanaSearcher } from './searcher';
+import { NestedFolderDTO } from './types';
+import { queryResultToViewItem } from './utils';
+
+export async function getFolderChildren(parentUid?: string, parentTitle?: string): Promise<DashboardViewItem[]> {
+  if (!config.featureToggles.nestedFolders) {
+    console.error('getFolderChildren requires nestedFolders feature toggle');
+    return [];
+  }
+
+  if (!parentUid) {
+    // We don't show dashboards at root in folder view yet - they're shown under a dummy 'general'
+    // folder that FolderView adds in
+    const folders = await getChildFolders();
+    return folders;
+  }
+
+  const searcher = getGrafanaSearcher();
+  const dashboardsResults = await searcher.search({
+    kind: ['dashboard'],
+    query: '*',
+    location: parentUid,
```

Comment:
I had a look at the searcher code, but I don't quite understand what `location` refers to. I assume it is the URL?

Reply:
location is basically the folder we're searching. It returns only dashboard results at that location.